Sources/GRPCCore/Call/Server/ServerStreamState.swift

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Synchronization
+
+@available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+public struct ServerStreamState: Sendable {
```

Comment:
For discussion: I don't like this name. 

The other idea I had was `ServerStreamContext` which is potentially better and aligns with `ServerContext`. Downside is that it hints at a slightly wide use.

Another option: rather than nesting `Events` have a `ServerStreamEvents` type which also has a computed property for `isRPCCancelled` (i.e. an `AsyncSequence` with extra API).